src/cloudHandler.js

```diff
@@ -37,6 +44,31 @@ var handler = function (params, success, failure) {
 };
 
 /**
+ * Get client id 
+ * 
+ * @param {*} payload object to add parameter 
+ */
+function getClientId(payload) {
```

Comment:
it might be better to make this function async, and also allow override for this function to allow it work with different cases.